### PR TITLE
Add Backrest

### PR DESCRIPTION
**Pull request by mcmikemn**
Implements #362

### CODE REVIEW

File: backrest/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+services:
+  backrest:
+    image: garethgeorge/backrest:latest
+    volumes:
+      - ./backrest/data:/data
+      - ./backrest/config:/config
+      - ./backrest/cache:/cache
+      - backrest-restores:/backrest-restores
```
Comment (EnigmaCurry) [inline backrest/docker-compose.yaml:8]: Shouldn't these only be in the docker-compose.instance.yaml and use BACKREST_USE_EXTERNAL_VOLUME and BACKREST_EXTERNAL_VOLUME vars?

Comment (mcmikemn) [reply backrest/docker-compose.yaml]: The `backrest-restores` volume is defined in `docker-compose.instance.yaml` as either a named volume or a bind mount, based on `BACKREST_USE_EXTERNAL_VOLUME`.

I realize the other 3 volumes were bind mounts so I changed them in [0ac8e3b](https://github.com/EnigmaCurry/d.rymcg.tech/pull/363/commits/0ac8e3bdb2cb3f2a11c8e6bce1f6ccf3ccca4f96) to named volumes.

File: backrest/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+services:
+  backrest:
+    image: garethgeorge/backrest:latest
+    volumes:
+      - data:/data
+      - config:/config
+      - cache:/cache
+      - backrest-restores:/backrest-restores
```
Comment (EnigmaCurry) [inline backrest/docker-compose.yaml:8]: `      - backrest-restores:/backrest-restores`

Won't this throw an error if use_external_volume is false? In the instance yaml it is blocked by this conditional, so the volume config might not exist in the final override.yaml

File: backrest/docker-compose.instance.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  cache:
+  #@ if use_external_volume:
+  backrest-restores:
+    driver: local
+    driver_opts:
+      type: none
+      o: bind
+      device: (@= upload_location @)
+  #@ end
+  #@ if not use_external_volume:
```
Comment (EnigmaCurry) [inline backrest/docker-compose.instance.yaml:84]: this should be an else block

Comment (mcmikemn) [reply backrest/docker-compose.instance.yaml]: Fixed in [4b210e7](https://github.com/EnigmaCurry/d.rymcg.tech/pull/363/commits/4b210e77065e19ae12dcfdcc19e9938ca142b9ce)

**Review by EnigmaCurry — APPROVED** [review]
LGTM. I added a warning at the top about the concern I had about it not managing the docker container lifecycle.

**Comment by mcmikemn** [comment]
> LGTM. I added a warning at the top about the concern I had about it not managing the docker container lifecycle.

Good idea. My mindset was for a GUI to manage existing restic repos, but of course Backrest has the ability to create new backups, so this is good to mention.

**Review by EnigmaCurry — DISMISSED** [review]

**Pull request merged**